packages/shared-ux/modal/tabbed/src/tabbed_modal.tsx

```diff
@@ -117,7 +117,7 @@ const TabbedModalInner: FC<ITabbedModalInner> = ({
       aria-labelledby={modalTitle}
     >
       <EuiModalHeader>
-        <EuiModalHeaderTitle>{modalTitle}</EuiModalHeaderTitle>
+        <EuiModalHeaderTitle id={modalTitle}>{modalTitle}</EuiModalHeaderTitle>
```

Comment:
I took your PR for another spin, and the Accessibility tree isn't wanting to recognize the `aria-describedby`. I think what's happening is the attribute is looking for an ID attribute instead of the text string in the H1. What do you think about doing something like this:

```diff
+ import { useGeneratedHtmlId } from '@elastic/eui';

+ const shareModalHeadingId = useGeneratedHtmlId();

<EuiModal
  onClose={() => {
  onClose();
    setTimeout(() => anchorElement?.focus(), 1);
  }}
  style={{ ...(modalWidth ? { width: modalWidth } : {}) }}
  maxWidth={true}
  data-test-subj="shareContextModal"
- aria-labelledby={modalTitle}
+ aria-labelledby={shareModalHeadingId}
>
...
<EuiModalHeaderTitle
- id={modalTitle}
+ id={shareModalHeadingId}
>{modalTitle}</EuiModalHeaderTitle>
```

I mocked it up quickly on my local machine and verified the expected accessible title was showing up in Chrome's Accessibility Tree. Hopefully that lets you make a change to the component without having to go through a reducer function. LMK if this won't work or will be overly difficult!